build.sbt

```diff
@@ -40,9 +40,24 @@ val commonSettings = Defaults.coreDefaultSettings ++ Seq(
     "-Ywarn-numeric-widen",
     "-Ywarn-value-discard",
     "-Xfuture",
-    "-Ywarn-unused",
-    "-Ywarn-unused-import" // 2.11 only
-  )
+    "-Ywarn-unused"
+  ) ++ ( //target:jvm-1.8 supported from 2.11.5, warn-unused-import deprecated in 2.12
+        if (scalaVersion.value startsWith "2.11") {
+          val revision = scalaVersion.value.split('.').last.toInt
+          Seq("-Ywarn-unused-import") ++ (
+            if (revision >= 5) {
+              Seq("-target:jvm-1.8")
+            }
+            else {
+              Nil
+            })
+        }
+        else Nil
+    )
+    ++ (
+        if (scalaVersion.value startsWith "2.12") Seq("-target:jvm-1.8") // what about "-Ypartial-unification" (SI-2712)?
```

Comment:
what of compiler versions exactly do you want to support? And which one should be assigned to scalaVersion?